### PR TITLE
[XLA:TSL] Skip thread NUMA binding when the node has no allowed CPUs

### LINKED ISSUE #45484: numa_hwloc.cc:121 Call to hwloc_set_cpubind() failed: Invalid argument
**System information:**

- uname: Linux 4.18.0-372.32.1.el8_6.x86_64
- OS name: Springdale Open Enterprise Linux
- OS version: 8.6 (Modena)
- Python version: 3.12.13
- JAX version: 0.10.2
- GPU model and memory: NVIDIA A100-SXM4-40GB

**Problem description:**

On a SLURM cluster, when initializing XLA, it spams the console with `ERROR` level logs. These originate from line 121 of `numa_hwloc.cc`.

Example:

```shell
$ srun --partition=gpu --gpus=1 python3 -c 'import jax; print(jax.devices())'
E0710 03:21:27.670090 2522223 numa_hwloc.cc:121] Call to hwloc_set_cpubind() failed: Invalid argument [22]
[CudaDevice(id=0)]
```

Because the cluster uses standard cgroups to restrict which CPU cores a SLURM job can access, XLA's attempt to bind threads via `hwloc_set_cpubind()` is rejected by the OS with `EINVAL` (Invalid argument [22]).

Since XLA gracefully falls back to the default OS thread scheduler and GPU execution continues normally, logging this as a fatal-looking `ERROR` is misleading.

**Proposed solution:**

Consider downgrading this specific `hwloc` binding failure from `LOG(ERROR)` to `VLOG(1)` or `LOG(WARNING)`. Alternatively, XLA could silently catch `EINVAL` during thread pinning, given that failing to pin a thread is a standard, non-fatal operational reality on modern HPC clusters.

### PR DESCRIPTION
**Pull request by kodlan**
## 📝 Summary of Changes

`NUMASetThreadNodeAffinity` now returns early (with a `VLOG(1)`) when the hwloc NUMA node object has an empty cpuset instead of calling `hwloc_set_cpubind` and logging an ERROR.

The topology is loaded without `HWLOC_TOPOLOGY_FLAG_INCLUDE_DISALLOWED`, so hwloc already intersects every object's cpuset with the cgroup-allowed set at load time. An empty cpuset therefore means the environment forbids every CPU of that node, and the bind is guaranteed to fail: hwloc rejects empty sets with `EINVAL` in `hwloc_fix_cpubind` before any syscall. Genuinely unexpected bind failures still log at ERROR as before.

## 🎯 Justification

Fixes #45484. On SLURM clusters the job cgroup often grants no CPUs on the GPU's NUMA node, so every process prints a fatal-looking `E ... numa_hwloc.cc:121] Call to hwloc_set_cpubind() failed: Invalid argument [22]` during startup even though XLA falls back to normal OS scheduling and works fine. Failing to pin a thread because of cluster policy is an expected condition, not an error. Before the NUMA code consolidation in October 2025 this failure was silently ignored, so the ERROR is not a long-standing contract.

Known related path, intentionally out of scope: if the cgroup also restricts the node's memory (`cpuset.mems`), hwloc drops the node object entirely and the "Could not find hwloc NUMA node" ERROR a few lines above fires instead. The issue reports only the cpubind message, so this change stays minimal and does not touch that path.

## 🚀 Kind of Contribution

🐛 Bug Fix

## 🧪 Unit Tests:

There is no existing test for `numa_hwloc.cc`, and the new branch depends on host cgroup state (the topology is a process-wide singleton over the real machine). A deterministic test would need a dedicated binary loading a crafted XML topology via `HWLOC_XMLFILE` before the first `tsl::port` call, plus log capture; that coupling to hwloc internals costs more than a 7-line log-noise fix justifies, so no unit test is added. The mechanism and the fix were verified against the vendored hwloc 2.7.1 with a standalone harness: a synthetic two-node topology restricted to one node's CPUs (the cgroup situation) keeps the other node's object alive with an empty cpuset, `hwloc_set_cpubind` on it fails with `EINVAL` (the exact reported error), and the guarded logic skips it. On nodes with any allowed CPU the guard is unreachable, so behavior on unrestricted hosts is unchanged. `//xla/tsl/platform:env_test` and `//xla/tsl/platform:threadpool_test` (which exercise the thread-start path that calls this function) pass.

## 🧪 Execution Tests:

Not applicable: logging-only change in host runtime code, no change to compiled programs or device execution.

### CODE REVIEW

**Review by majnemer — APPROVED** [review]
LGTM
